redbot/setup.py

```diff
@@ -143,7 +153,9 @@ def get_name(name: str) -> str:
             " A-z, numbers, underscores (_) and periods (.)."
         )
         name = input("> ")
-        if INSTANCE_NAME_RE.fullmatch(name) is None:
+        if not name:
+            pass
+        elif INSTANCE_NAME_RE.fullmatch(name) is None:
             print(
                 "ERROR: Instance names can only include characters A-z, numbers, "
                 "underscores (_) and periods (.)."
```

Comment:
This message (and possibly the opening message) should be expanded to include the additional limitations this PR adds, so a user is not confused as to why their name is rejected.